src/main/java/codes/ztereohype/ztereomusic/command/commands/Play.java

```diff
@@ -49,20 +53,17 @@ public void execute(MessageReceivedEvent messageEvent, String[] args) {
 
         // check if args merged are/have url, if so try to feed it into lava, else try to youtube api the fuck out of it.
         String mergedArgs = String.join(" ", args);
-
-        Pattern urlPattern = Pattern.compile("^(http|https)://([a-z]+\\.[a-z]+)+/\\S+$", Pattern.CASE_INSENSITIVE);
-        Matcher matchedUrls = urlPattern.matcher(mergedArgs);
+        Matcher matchedUrls = URL_PATTERN.matcher(mergedArgs);
         boolean urlFound = matchedUrls.find();
 
-        // todo: clean up this bullshit try catch thing
         String identifier;
         if (!urlFound) {
-            // youtube api shit
-            try {
-                identifier = YoutubeSearch.getVideoUrl(mergedArgs);
-            } catch (IOException e) {
-                e.printStackTrace();
-                identifier = "error";
+            Optional<String> query = YoutubeSearch.query(mergedArgs);
```

Comment:
the more i learn about java the more i wanna learn O_O
the optional type is sick